### PR TITLE
{compiler}[GCCcore/10.2.0,gcccuda/2020b] AOMP v13.0.2

### PR DESCRIPTION
**Pull request by nordmoen**
This is an updated version of the PR #12018. This PR updates the version
to `13.0-2`. In addition this PR adds an improved patch which should
force the built modules to pick up the EasyBuild installed version of
`gcc`.

This module requires the [`aomp.py` EasyBlock](https://github.com/easybuilders/easybuild-easyblocks/pull/2435) to build.

### CODE REVIEW

**Comment by nordmoen** [comment]
@migueldiascosta Could you test if the patch included here is enough to avoid the problems of #12018 and confusion between which `gcc` is picked up with `LLVM`?

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2435
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
sms - Linux centos linux 7.6.1810, x86_64, AMD EPYC 7601 32-Core Processor (zen), Python 2.7.5
See https://gist.github.com/399377ad4ec4ba947777c035a8b71d3f for a full test report.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2435
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
sms - Linux centos linux 7.6.1810, x86_64, AMD EPYC 7601 32-Core Processor (zen), Python 2.7.5
See https://gist.github.com/2f15bdfaf84f4b90388ab093122777d9 for a full test report.

**Comment by migueldiascosta** [comment]
the earlier problems are solved, but the symbolic link creates problems when reinstalling ( `OSError('Cannot call rmtree on a symbolic link',)`)

(note: the tests above don't include the cuda variant)

**Comment by nordmoen** [comment]
> @nordmoen the earlier problems are solved, but the symbolic link creates problems when reinstalling ( `OSError('Cannot call rmtree on a symbolic link',)`)
> 
> (note: the tests above don't include the cuda variant)

I tried to patch this out, but it immediately fails in other parts of the build. Any tips on workarounds?

**Comment by migueldiascosta** [comment]
@nordmoen so, is it the `build_aomp.sh` script that creates the link? maybe the easyblock can then cleanup by removing the link and renaming the directory?

**Comment by nordmoen** [comment]
> @nordmoen so, is it the `build_aomp.sh` script that creates the link? maybe the easyblock can then cleanup by removing the link and renaming the directory?

It seems to be the `build_project.sh` script which is called from `build_aomp.sh` (and is required). I will look into removing it in the EasyBlock.

**Comment by nordmoen** [comment]
@migueldiascosta I updated the EasyBlock which seems to work locally. Could you test again when you have the time?

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2435
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
sms - Linux centos linux 7.6.1810, x86_64, AMD EPYC 7601 32-Core Processor (zen), Python 2.7.5
See https://gist.github.com/51c93914c3f152bc00fe8364ca2a0611 for a full test report.

**Comment by nordmoen** [comment]
Now that the `aomp.py` easyblock has been merged, could someone re-run these checks?

**Comment by akesandgren** [comment]
You can do that by doing, "eb --sync-pr-with-develop 12909"

**Comment by nordmoen** [comment]
> You can do that by doing, "eb --sync-pr-with-develop 12909"

Thanks @akesandgren !

File: easybuild/easyconfigs/a/AOMP/AOMP-13.0-2-GCCcore-10.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+version = '13.0-2'
+
+homepage = 'https://github.com/ROCm-Developer-Tools/aomp'
+description = "AMD fork of LLVM, setup for OpenMP offloading to Accelerators"
+
+toolchain = {'name': 'GCCcore', 'version': '10.2.0'}
+
+source_urls = ['https://github.com/ROCm-Developer-Tools/aomp/releases/download/rel_%(version)s/']
+sources = ['aomp-%(version)s.tar.gz']
+patches = ['build_aomp-%(version)s.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/a/AOMP/AOMP-13.0-2-GCCcore-10.2.0.eb:19]: Please rename the patch file to something like `AOMP-13.0.2_short-description.patch`

File: easybuild/easyconfigs/a/AOMP/AOMP-13.0-2-GCCcore-10.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+dependencies = [
+    ('binutils', '2.35'),
+    ('libffi', '3.3'),
+    ('ncurses', '6.2'),
+    ('numactl', '2.0.13'),
+]
+
+sanity_check_paths = {
+    'files': ['bin/clang', 'bin/ld.lld', 'include/omp.h'],
+    'dirs': ['amdgcn', 'bin', 'include', 'lib'],
```
Comment (boegel) [inline easybuild/easyconfigs/a/AOMP/AOMP-13.0-2-GCCcore-10.2.0.eb:42]: Ideally, this is done in the AOMP easyblock, and a custom sanity check command is added too?

It also helps (humans) to keep the entries alphabetically sorted.

File: easybuild/easyconfigs/a/AOMP/build_aomp-13.0-2.patch [hunk 3]
```diff
@@ -0,0 +1,218 @@
+--- a/aomp/bin/aomp_common_vars
```
Comment (boegel) [inline easybuild/easyconfigs/a/AOMP/build_aomp-13.0-2.patch:1]: Please add a short description on top of the patch file, and mention where the patch comes from, or who created it (and keep in mind that this implies changing the SHA256 checksum in the easyconfig file)

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/a/AOMP/build_aomp-13.0-2.patch (hunk 3 above)
Comment (nordmoen) [reply easybuild/easyconfigs/a/AOMP/build_aomp-13.0-2.patch]: Fixed

File: easybuild/easyconfigs/a/AOMP/AOMP-13.0-2-GCCcore-10.2.0.eb (hunk 1 above)
Comment (nordmoen) [reply easybuild/easyconfigs/a/AOMP/AOMP-13.0-2-GCCcore-10.2.0.eb]: Fixed

File: easybuild/easyconfigs/a/AOMP/AOMP-13.0-2-GCCcore-10.2.0.eb (hunk 2 above)
Comment (nordmoen) [reply easybuild/easyconfigs/a/AOMP/AOMP-13.0-2-GCCcore-10.2.0.eb:44]: Updated with easybuilders/easybuild-easyblocks#2464

**Comment by smoors** [comment]
@boegelbot please test @ generoso EB_ARGS="--include-easyblocks-from-pr 2464"

**Comment by smoors** [comment]
Test report by @smoors
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2464
**FAILED**
Build succeeded for 3 out of 5 (2 easyconfigs in total)
node381.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/b84843f7b5eab15093dc057f90ff9d63 for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2464
**FAILED**
Build succeeded for 4 out of 5 (2 easyconfigs in total)
node153.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz (ivybridge), Python 2.7.5
See https://gist.github.com/f278102f87f3c6c272fca19f476c1634 for a full test report.

**Comment by nordmoen** [comment]
> Test report by @smoors
> Using easyblocks from PR(s) [easybuilders/easybuild-easyblocks#2464](https://github.com/easybuilders/easybuild-easyblocks/pull/2464)
> **FAILED**
> Build succeeded for 4 out of 5 (2 easyconfigs in total)
> node153.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz (ivybridge), Python 2.7.5
> See https://gist.github.com/f278102f87f3c6c272fca19f476c1634 for a full test report.

@smoors To build the AOMP module with CUDA support you have to supply `--cuda-compute-capabilities`

**Comment by smoors** [comment]
Test report by @smoors
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2464
**SUCCESS**
Build succeeded for 5 out of 5 (2 easyconfigs in total)
node153.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz (ivybridge), Python 2.7.5
See https://gist.github.com/f30ba6582420c1a43b4120611657b118 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS="--cuda-compute-capabilities 7.0"
CORE_CNT=16

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (2 easyconfigs in total)
node3301.joltik.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), Python 3.6.8
See https://gist.github.com/9280af96a8f88c7f5c1719c2bb0d7ef3 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @nordmoen!

**Pull request merged**